NeoML/src/Dnn/Layers/CastLayer.cpp

```diff
@@ -69,7 +69,6 @@ void CCastLayer::RunOnce()
 
 void CCastLayer::BackwardOnce()
 {
-	NeoAssert( inputBlobs[0]->GetDataType() == CT_Float && outputBlobs[0]->GetDataType() == CT_Float );
```

Comment:
May be use diff blobs?

Reply:
Fixed by moving check to `::Reshape()` (which is more optimal)